clients/src/main/java/org/apache/kafka/common/protocol/types/ArrayOf.java

```diff
@@ -72,6 +94,9 @@ public String toString() {
     @Override
     public Object[] validate(Object item) {
         try {
+            if (isNullable() && item == null)
+                return null;
```

Comment:
Yeah, I saw that. I'm suggesting a separate SchemaException with message specific for nulls, to make troubleshooting/debugging a bit easier. Not a big deal though.